app/templates/buckets/details.html

```diff
@@ -7,47 +7,78 @@
-          <td>{{ yes_no(users3bucket.access_level, 'readwrite') }}</td>
-          <td class="align-right">
-            {% if bucket.has_admin(current_user) or current_user.is_superuser %}
-              <!-- Button to change access level -->
-              <form action="{{ url_for('users3buckets.update', { id: users3bucket.id }) }}" method="post" class="inline-form clearfix">
-                <input type="hidden" name="access_level" value="{{ yes_no(users3bucket.access_level, 'readwrite', 'readonly', 'readwrite') }}" />
-
-                <input type="hidden" name="redirect_to" value="{{ url_for('buckets.details', { id: bucket.id }) }}" />
-
-                <input type="submit" class="js-confirm button button-secondary" value="{{ yes_no(users3bucket.access_level, 'readwrite', 'Revoke', 'Grant') }} read/write access" />
-              </form>
-
-              <!-- Button to revoke access -->
-              <form action="{{ url_for('users3buckets.delete', { id: users3bucket.id }) }}" method="post" class="inline-form clearfix">
-                <input type="hidden" name="redirect_to" value="{{ url_for('buckets.details', { id: bucket.id }) }}">
-
-                <input type="submit" class="js-confirm button button-secondary" value="Revoke admin access" />
-              </form>
+          <td><a class="{% if current_user.auth0_id == users3bucket.user.auth0_id %}highlight-current{% endif %}" href="{{ url_for('users.details', { id: users3bucket.user.auth0_id }) }}">{% if users3bucket.user.name %}{{ users3bucket.user.name }} {% endif %}({{ users3bucket.user.username }})</a></td>
+          <td>
+            {% if(users3bucket.is_admin) %}
+              admin
+            {% else %}
+              {{ users3bucket.access_level }}
             {% endif %}
           </td>
+          {% if current_user_is_bucket_admin or current_user.is_superuser %}
+            <td class="align-right">
+
+              <div class="form-group change-data-access-level-panel panel panel-border-narrow js-hidden" id="users3bucket_{{ users3bucket.id }}_change-data-access-level-panel">
+
+                <button class="js-close-panel">Close panel</button>
+
+                <form action="{{ url_for('users3buckets.update', { id: users3bucket.id }) }}" method="post" class="inline-form clearfix">
+                  <input type="hidden" name="redirect_to" value="{{ url_for('buckets.details', { id: bucket.id }) }}" />
+                  <div class="form-group">
+                    <fieldset>
+                      <legend class="visuallyhidden">Data access level</legend>
+                      <div class="multiple-choice">
+                        <input type="radio" id="users3bucket_{{ users3bucket.id }}_data_access_level-readonly" name="users3bucket_{{ users3bucket.id }}_data_access_level" value="readonly" {% if users3bucket.access_level == 'readonly' %}checked{% endif %}>
```

Comment:
If these are all separate forms, you could remove the `{{ users3bucket.id }}` from the `name` attribute and simplify the handler code.

Reply:
Not from the `id` though, because unique.